### PR TITLE
SessionServerAppManager refactoring

### PR DESCRIPTION
**Pull request by skywojciechowskim**
Summary: SessionServerAppManager refactoring
Type: Feature
Test Plan: UT
Jira: RIALTO-4

### CODE REVIEW

**Review by skypetts — DISMISSED** [review]
Looks OK to me but needs a thorough review from someone else too please!

File: serverManager/common/source/SessionServerAppManager.cpp [hunk 1]
```diff
@@ -109,200 +112,116 @@ void SessionServerAppManager::onSessionServerStateChanged(int serverId,
...
+bool SessionServerAppManager::configureSessionServer(const std::unique_ptr<ISessionServerApp> &sessionServer)
 {
-    std::unique_lock<std::mutex> lock{m_sessionServerAppsMutex};
-    auto app = std::find_if(m_sessionServerApps.begin(), m_sessionServerApps.end(),
-                            [&](const auto &app) { return app->getId() == serverId; });
-    if (app == m_sessionServerApps.end())
+    if (!sessionServer)
     {
-        RIALTO_SERVER_MANAGER_LOG_ERROR("Configuration of app with id: %d failed - app not found", serverId);
+        RIALTO_SERVER_MANAGER_LOG_ERROR("Configuration of app with id: %d failed - app not found",
```
Comment (aczs) [inline serverManager/common/source/SessionServerAppManager.cpp:174]: Change "app" to "server"?

File: serverManager/common/source/SessionServerAppManager.cpp [hunk 2]
```diff
@@ -109,200 +112,116 @@ void SessionServerAppManager::onSessionServerStateChanged(int serverId,
...
-                                                                         (*app)->getMaxWebAudioPlayers()};
-    if (!m_ipcController->performSetConfiguration((*app)->getId(), initialState, socketName, maxResource))
+    const auto initialState{sessionServer->getInitialState()};
+    const auto socketName{sessionServer->getSessionManagementSocketName()};
+    const firebolt::rialto::common::MaxResourceCapabilitites maxResource{sessionServer->getMaxPlaybackSessions(),
+                                                                         sessionServer->getMaxWebAudioPlayers()};
+    if (!m_ipcController->performSetConfiguration(sessionServer->getServerId(), initialState, socketName, maxResource))
     {
-        RIALTO_SERVER_MANAGER_LOG_ERROR("Configuration of app with id %d failed - ipc error.", serverId);
+        RIALTO_SERVER_MANAGER_LOG_ERROR("Configuration of app with id %d failed - ipc error.",
```
Comment (aczs) [inline serverManager/common/source/SessionServerAppManager.cpp:184]: Change "app" to "server"?

File: serverManager/common/source/SessionServerAppManager.cpp [hunk 3]
```diff
@@ -109,200 +112,116 @@ void SessionServerAppManager::onSessionServerStateChanged(int serverId,
...
+                                        sessionServer->getServerId());
         return false;
     }
-    RIALTO_SERVER_MANAGER_LOG_INFO("Configuration of app with id %d succeeded.", serverId);
+    RIALTO_SERVER_MANAGER_LOG_INFO("Configuration of app with id %d succeeded.", sessionServer->getServerId());
```
Comment (aczs) [inline serverManager/common/source/SessionServerAppManager.cpp:188]: Change "app" to "server"?

File: serverManager/common/source/SessionServerAppManager.cpp [hunk 4]
```diff
@@ -109,200 +112,116 @@ void SessionServerAppManager::onSessionServerStateChanged(int serverId,
...
 {
-    std::unique_lock<std::mutex> lock{m_sessionServerAppsMutex};
-    auto app = std::find_if(m_sessionServerApps.begin(), m_sessionServerApps.end(),
-                            [&](const auto &app) { return app->getId() == serverId; });
-    if (app == m_sessionServerApps.end())
+    if (!sessionServer)
     {
-        RIALTO_SERVER_MANAGER_LOG_ERROR("Configuration of app with id: %d failed - app not found", serverId);
+        RIALTO_SERVER_MANAGER_LOG_ERROR("Configuration of app with id: %d failed - app not found",
+                                        sessionServer->getServerId());
```
Comment (aczs) [inline serverManager/common/source/SessionServerAppManager.cpp:175]: You are dereferencing null here.

File: serverManager/common/source/SessionServerAppManager.cpp (hunk 1 above)
Comment (skywojciechowskim) [reply serverManager/common/source/SessionServerAppManager.cpp:174]: done

File: serverManager/common/source/SessionServerAppManager.cpp (hunk 4 above)
Comment (skywojciechowskim) [reply serverManager/common/source/SessionServerAppManager.cpp:175]: fixed

File: serverManager/common/source/SessionServerAppManager.cpp (hunk 2 above)
Comment (skywojciechowskim) [reply serverManager/common/source/SessionServerAppManager.cpp:184]: changed

File: serverManager/common/source/SessionServerAppManager.cpp (hunk 3 above)
Comment (skywojciechowskim) [reply serverManager/common/source/SessionServerAppManager.cpp:188]: changed

**Review by aczs — APPROVED** [review]

**Review by skypetts — APPROVED** [review]

**Pull request merged**